### DISCUSSION TITLE
package.json: conditional exports: deno export

### DISCUSSION THREAD

**Discussion by balupton** [denoland/deno#17964] [Ideas]
I couldn't find any documentation about this on:

- https://deno.com/blog/v1.31#packagejson-support
- https://deno.land/manual@v1.31.1/node/package_json

Node.js now supports a feature called conditional exports:

- https://nodejs.org/api/packages.html#conditional-exports

Would be cool if there was support for a `deno` export, such that I could write code that specifically has a different export for Node `node`, Deno `deno`, Browsers `import`, and legacy Node `require`/`default`

For instance, my [`start-of-week` package](https://github.com/bevry/start-of-week) writes different code for each target, so it would definitely use this. Currently it is using [boundation](https://github.com/bevry/boundation) with [editions](https://github.com/bevry/editions) to achieve the conditional loading, however now that Node.js has its own official convention, I am planning to adopt that instead for all my packages.

**Comment by balupton** [comment]
I did find this however:

- https://runtime-keys.proposal.wintercg.org/#deno

Perhaps it is supported? but not documented anywhere officially?

**Comment by bartlomieju** [comment]
Yes, this has been supported for several releases already - you can use `deno` condition. I believe this was in the manual, but somehow dissappeared.

**Reply by kigawas** [reply]
It seems here: https://deno.land/x/manual@v1.21.0/node/compatibility_mode.md?source=#module-resolution

But it does not work on deno 2 (seemingly?)

**Reply by kigawas** [reply]
@bartlomieju @balupton 

Alright I see. When there is a package A with exports deno condition, if we import A directly with deno, deno export works. If we import package B using package A, deno export will not work

**Reply by yacinehmito** [reply]
> If we import package B using package A, deno export will not work

Is there an issue for that?

**Reply by kigawas** [reply]
@yacinehmito Related issue: https://github.com/denoland/deno/issues/23757
